src/ol/style/expressions.js

```diff
@@ -0,0 +1,612 @@
+  if (!(getValueType(value) & ValueTypes.NUMBER)) {
+    throw new Error(`A numeric value was expected, got ${JSON.stringify(value)} instead`);
+  }
+}
+function assertString(value) {
+  if (!(getValueType(value) & ValueTypes.STRING)) {
+    throw new Error(`A string value was expected, got ${JSON.stringify(value)} instead`);
+  }
+}
+function assertBoolean(value) {
+  if (!(getValueType(value) & ValueTypes.BOOLEAN)) {
+    throw new Error(`A boolean value was expected, got ${JSON.stringify(value)} instead`);
+  }
+}
+function assertArgsCount(args, count) {
+  if (args.length !== count) {
+    throw new Error(`Exactly ${count} arguments were expected, got ${args.length} instead`);
+  }
+}
+function assertArgsMinCount(args, count) {
+  if (args.length < count) {
+    throw new Error(`At least ${count} arguments were expected, got ${args.length} instead`);
+  }
+}
+function assertArgsMaxCount(args, count) {
+  if (args.length > count) {
+    throw new Error(`At most ${count} arguments were expected, got ${args.length} instead`);
+  }
+}
+function assertArgsEven(args) {
+  if (args.length % 2 !== 0) {
+    throw new Error(`An even amount of arguments was expected, got ${args} instead`);
+  }
+}
+function assertUniqueInferredType(args, types) {
+  if (!isTypeUnique(types)) {
+    throw new Error(`Could not infer only one type from the following expression: ${JSON.stringify(args)}`);
+  }
+}
+
+Operators['get'] = {
+  getReturnType: function(args) {
+    return ValueTypes.ANY;
+  },
+  toGlsl: function(context, args) {
+    assertArgsCount(args, 1);
+    assertString(args[0]);
+    const value = args[0].toString();
+    if (context.attributes.indexOf(value) === -1) {
+      context.attributes.push(value);
+    }
+    const prefix = context.inFragmentShader ? 'v_' : 'a_';
+    return prefix + value;
+  }
+};
+Operators['var'] = {
+  getReturnType: function(args) {
+    return ValueTypes.ANY;
+  },
+  toGlsl: function(context, args) {
+    assertArgsCount(args, 1);
+    assertString(args[0]);
+    const value = args[0].toString();
+    if (context.variables.indexOf(value) === -1) {
+      context.variables.push(value);
+    }
+    return `u_${value}`;
+  }
+};
+Operators['time'] = {
+  getReturnType: function(args) {
+    return ValueTypes.NUMBER;
+  },
+  toGlsl: function(context, args) {
+    assertArgsCount(args, 0);
+    return 'u_time';
+  }
+};
+Operators['zoom'] = {
+  getReturnType: function(args) {
+    return ValueTypes.NUMBER;
+  },
+  toGlsl: function(context, args) {
+    assertArgsCount(args, 0);
+    return 'u_zoom';
+  }
+};
+Operators['resolution'] = {
+  getReturnType: function(args) {
+    return ValueTypes.NUMBER;
+  },
+  toGlsl: function(context, args) {
+    assertArgsCount(args, 0);
+    return 'u_resolution';
+  }
+};
+Operators['*'] = {
+  getReturnType: function(args) {
+    return ValueTypes.NUMBER;
+  },
+  toGlsl: function(context, args) {
+    assertArgsCount(args, 2);
+    assertNumber(args[0]);
```

Comment:
See https://github.com/fredj/openlayers/commit/94127ca28adf9fff33e2574c3e4bef6a2872aa4d for a small simplification: I've added a `assertNumbers` function that takes an array and check that every item is a number.

Reply:
Well done, I'll add this commit to the PR